### PR TITLE
Pass texture table to wall.register function.

### PR DESCRIPTION
**Pull request by kakalak-lumberJack**
Registration function will take texture table instead of individual texture. Would break registrations by other mods.

### CODE REVIEW

**Comment by rubenwardy** [comment]
It would be pretty easy to make this backwards compatible.

```lua
if typeof(wall_texture_table) ~= "table" then
    texture_table = { wall_texture_table }
end
```

**Comment by kakalak-lumberJack** [comment]
Brilliant. Fixed

File: mods/walls/init.lua [hunk 1]
```diff
@@ -1,6 +1,10 @@
 walls = {}
 
-walls.register = function(wall_name, wall_desc, wall_texture, wall_mat, wall_sounds)
+walls.register = function(wall_name, wall_desc, wall_texture_table, wall_mat, wall_sounds)
+	--make wall_texture_table paramenter backwards compatible for mods passing single texture
+	if type(wall_texture_table) ~= "table" then
```
Comment (SmallJoker) [inline mods/walls/init.lua]: Use tabulators to indent.

File: mods/walls/init.lua [hunk 2]
```diff
@@ -35,12 +39,12 @@ walls.register = function(wall_name, wall_desc, wall_texture, wall_mat, wall_sou
...
-walls.register("walls:cobble", "Cobblestone Wall", "default_cobble.png",
+walls.register("walls:cobble", "Cobblestone Wall", {"default_cobble.png"},
 		"default:cobble", default.node_sound_stone_defaults())
 
-walls.register("walls:mossycobble", "Mossy Cobblestone Wall", "default_mossycobble.png",
+walls.register("walls:mossycobble", "Mossy Cobblestone Wall", {"default_mossycobble.png"},
 		"default:mossycobble", default.node_sound_stone_defaults())
 
-walls.register("walls:desertcobble", "Desert Cobblestone Wall", "default_desert_cobble.png",
+walls.register("walls:desertcobble", "Desert Cobblestone Wall", {"default_desert_cobble.png"},
```
Comment (SmallJoker) [inline mods/walls/init.lua]: These changes are now no longer required.

Comment (kakalak-lumberJack) [reply mods/walls/init.lua]: Agreed, not required but might be useful as an example.

File: mods/walls/init.lua (hunk 1 above)
Comment (kakalak-lumberJack) [reply mods/walls/init.lua]: Done.

**Pull request merged**